src/python/pants/build_graph/address.py

```diff
@@ -514,10 +514,7 @@ def sanitize(s: str) -> str:
         else:
             target = f"{parent_prefix}{self.target_name}"
         if self.parameters:
-            key_value_strs = ",".join(
-                f"{sanitize(k)}={sanitize(v)}" for k, v in self.parameters.items()
-            )
-            params = f"@@{key_value_strs}"
+            params = f"@{sanitize(self.parameters_repr)}"
```

Comment:
This is unrelated to the refactor, but it does make `path_safe_spec` slightly easier to read